src/osd/mClockClientProfileQueue.h

```diff
@@ -0,0 +1,148 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2016 Red Hat Inc.
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+
+#pragma once
+
+#include <ostream>
+
+#include "boost/variant.hpp"
+
+#include "common/config.h"
+#include "common/ceph_context.h"
+#include "common/mClockPriorityQueue.h"
+#include "osd/OpQueueItem.h"
+#include "osd/mClockOpClassSupport.h"
+#include "common/mClockClientInfoMgr.h"
+
+
+namespace ceph {
+
+  using Request = OpQueueItem;
+  using Client = uint64_t;
+  using MClockProfile = uint64_t;
+
+  // This class exists to bridge the ceph code, which treats the class
+  // as the client, and the queue, where the class is
+  // osd_op_type_t. So this adapter class will transform calls
+  // appropriately.
+  class mClockClientProfileQueue : public OpQueue<Request, Client> {
+
+  public:
+
+    // either wraps a request or acts as a proxy for a client request,
+    // the actual request of which is in client_queue.
+    struct RequestProxyWrapper {
+      std::unique_ptr<Request> request;
+      bool is_proxy;
+
+      // construct a proxy
+      RequestProxyWrapper() :
+	is_proxy(true)
+      {}
+
+      // construct a request wrapper
+      explicit RequestProxyWrapper(Request&& _request) :
+	request(new Request(std::move(_request))),
+	is_proxy(false)
+      {}
+
+      RequestProxyWrapper(RequestProxyWrapper&& _wrapper) {
+	std::swap(request, _wrapper.request);
+	std::swap(is_proxy, _wrapper.is_proxy);
```

Comment:
`is_proxy` is uninitialized in this constructor

the compiler will generate these move operations for you though, so you should be safe to remove them altogether? see https://godbolt.org/g/zqZUtz for a working example